src/game_battler.cpp

```diff
@@ -78,12 +79,28 @@ RPG::State::Restriction Game_Battler::GetSignificantRestriction() const {
 }
 
 bool Game_Battler::CanAct() const {
-	const std::vector<int16_t> states = GetInflictedStates();
-	for (int i = 0; i < (int)states.size(); i++) {
-		// States are guaranteed to be valid
-		const RPG::State* state = ReaderUtil::GetElement(Data::states, states[i]);
-		if (state->restriction == RPG::State::Restriction_do_nothing) {
-			return false;
+	const auto& states = GetStates();
+	for (size_t i = 0; i < states.size(); ++i) {
+		if (states[i] > 0) {
+			const auto* state = ReaderUtil::GetElement(Data::states, i + 1);
+			assert(state);
+			if (state->restriction == RPG::State::Restriction_do_nothing) {
+				return false;
+			}
+		}
+	}
+	return true;
+}
+
+bool Game_Battler::CanActOrRecoverable() const {
+	const auto& states = GetStates();
+	for (size_t i = 0; i < states.size(); ++i) {
+		if (states[i] > 0) {
+			const auto* state = ReaderUtil::GetElement(Data::states, i + 1);
+			assert(state);
+			if (state->restriction == RPG::State::Restriction_do_nothing && state->auto_release_prob == 0) {
```

Comment:
After sleeping about this the function makes sense: The idea is to handle "Can act right now or will be able to act again without external help in the future".